File: 2020-TAMUCTF/PWN/TROLL/troll_crack.py
import random
import time
random.seed(time.time())

for x in range(0, 0x63):
    z = x = random.randint()

    x *= 0x14f8b589
    x //= 2**0x20

    y = x

    y //= 2**0xD

    y *= 0x186a0

    j = z - y
    j += 1

    print(j)

File: 2020-TAMUCTF/PWN/TROLL/troll_crack.c
#include <stdlib.h>
#include <stdio.h>
#include <math.h>
#include <time.h>

int main()
{
    sleep(2);
    int j = time(NULL);

    printf("he\n");
    srand(j);
    for (int i = 0; i <= 0x63; i++)
    {

        int z = rand();

        long int x = z;

        x *= 0x14f8b589;
        x /= pow(2, 0x20);

        int y = x;

        y /= pow(2, 0xD);

        y *= 0x186a0;

        int j = z - y;
        j += 1;

        printf("%d\n", j);
    }
}
